packages/upload-api/src/utils/rate-limits.js

```diff
@@ -0,0 +1,34 @@
+/**
+ *
+ * @param {import("../types").RateLimit} rateLimit
+ */
+const isBlock = (rateLimit) => rateLimit.rate === 0
+
+/**
+ *
+ * @param {import("../types").RateLimit[]} rateLimits
+ */
+const areAnyBlocks = (rateLimits) => rateLimits.some(isBlock)
+
+/**
+ * Query rate limits storage and find out if any of the given subjects
+ * have rate set to 0
+ *
+ * @param {import("../types").RateLimitsStorage} storage
+ * @param {string[]} subjects
+ * @return {Promise<import("@ucanto/interface").Result<boolean, import("@ucanto/interface").Failure>>}
+ */
+export async function areAnyBlocked(storage, subjects) {
```

Comment:
I would instead have something like `moderateRateLimit` and make it either succeed er error as opposed to erroring or returning a boolean.

Reply:
ah yea I ran with this and changed it to a `ensureRateLimit` function that also accepts a limit - this is more generally useful too which is nice, great idea!